### PR TITLE
[v2.0.0] Initial v2.0.0 commit - preparing repo for rewrite

### PR DESCRIPTION
**Pull request by phamann**
**⚠️ Note this is a PR into the `v.2.0.0` branch and not `master`**

### TL;DR
Prepares the repository for the v2.0.0 rewrite. Deletes all source, tests and dependencies from the tree and only leaves the README, LICENSE, CHANGELOG and package.json intact.

### Why?
The internal Fastly API that powers the Insights.js client and its tasks has changed. Therefore, we need to refactor the client to consume the new API. However, we have decided to take this opportunity to rewrite the client as a whole to ensure it meets todays requirements, make it more efficient and a better third-party citizen. 

This PR is the beginning of this work. We will submit small incremental feature pull requests using the `v.2.0.0` branch as the base to develop the new client. Eventually when we are happy with the new version we will merge it into `master`.

### Note
**The rewrite is not changing any of the underlying behaviour or functionality of the client that is already documented in the README. If we do change any behaviour it will be communicated and documented.** 

Most of the rewrite is to move the execution processing to work inside a WebWorker and therefore be more performant for the host websites and move to TypeScript to ensure better type safety and security.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,65 +4,10 @@
...
-    "clean": "rm -rf dist",
-    "build-development": "npm run clean && ./bin/build src/scout.js src/lib.js -v $npm_package_version",
-    "build-production": "npm run clean && ./bin/build src/scout.js src/lib.js -p -v $npm_package_version",
-    "build": "npm run verify && npm run build-development",
-    "format": "prettier --write \"**/*.js\"",
-    "lint": "eslint src test",
-    "verify": "npm run format && npm run lint",
-    "unit-test": "karma start --single-run",
-    "test": "npm run build && npm run unit-test"
+    "test": ""
```
Comment (clight-fastly) [inline package.json]: Nit: `test` needs to do _something_ or yarn doesn't like it:
`error An unexpected error occurred: "Script must exist".`
